Bugfix/expandable folder status model

claucambra opened the pull request:
Two things in this PR:

1. Code cleanup in general across `FolderStatusModel`
2. Fetch the first level of folders from root, fixing "fake" expandable folders

Fixes following bug:


https://github.com/nextcloud/desktop/assets/70155116/2c1868be-3c12-4982-a072-ec8d666746c8



<!-- 
Thanks for opening a pull request on the Nextcloud desktop client.

Instead of a Contributor License Agreement (CLA) we use a Developer Certificate of Origin (DCO).
https://en.wikipedia.org/wiki/Developer_Certificate_of_Origin

To accept that DCO, please make sure that you add a line like
Signed-off-by: Random Developer <random@developer.example.org>
at the end of each commit message.

This Signed-off-by trailer can be added automatically by git if you pass --signoff or -s to git commit.
See also:
https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---no-signoff
-->

claucambra commented on src/gui/folderstatusmodel.cpp:
```diff
@@ -1321,10 +1321,10 @@ void FolderStatusModel::slotShowFetchProgress()
+            const auto info = infoForIndex(idx);
             if (info && info->_fetchingJob) {
-                bool add = !info->hasLabel();
+                auto add = !info->hasLabel();
```
Thanks, changed

claucambra commented on src/gui/folderstatusmodel.cpp:
```diff
@@ -955,19 +960,19 @@ void FolderStatusModel::slotApplySelectiveSync()
-    auto par = qobject_cast<QWidget *>(QObject::parent());
-    if (!par->isVisible()) {
+    const auto parent = qobject_cast<QWidget *>(QObject::parent());
+    if (!parent->isVisible()) {
```
Good catch, now is checked

claucambra commented on src/gui/folderstatusmodel.cpp:
```diff
@@ -833,19 +838,19 @@ void FolderStatusModel::slotUpdateDirectories(const QStringList &list)
-        auto error = r->error();
+        qCDebug(lcFolderStatus) << reply->errorString();
+        parentInfo->_lastErrorString = reply->errorString();
+        auto error = reply->error();
```
Good point, I've removed it

claucambra commented on src/gui/folderstatusmodel.cpp:
```diff
@@ -195,86 +200,85 @@ QVariant FolderStatusModel::data(const QModelIndex &index, int role) const
+            const auto folder = subfolderInfo._folder;
+            if (!folder)
+                return {};
+            return {folder->path() + subfolderInfo._path};
```
Yup, it's a nicer way to do it here; changed

claucambra commented on src/gui/folderstatusmodel.cpp:
```diff
@@ -137,23 +145,20 @@ Qt::ItemFlags FolderStatusModel::flags(const QModelIndex &index) const
-
-    if (role == Qt::EditRole)
-        return QVariant();
+    if (!index.isValid() || (role == Qt::EditRole))
```
I've now added braces for all the if statements in the class

claucambra commented on src/gui/folderstatusmodel.cpp:
```diff
@@ -114,8 +123,7 @@ Qt::ItemFlags FolderStatusModel::flags(const QModelIndex &index) const
     case AddButton: {
-        Qt::ItemFlags ret;
-        ret = Qt::ItemNeverHasChildren;
+        auto ret = Qt::ItemNeverHasChildren;
```
Thanks, fixed

allexzander reviewed (changes requested):
Great. Kindly, fix the CI compile error and squash whatever makes sense (at least review comment fixes into one).

claucambra commented:
> Great. Kindly, fix the CI compile error and squash whatever makes sense (at least review comment fixes into one).

Thanks, fixed the CI error and squashed some commits

claucambra commented:
/backport to stable-3.12
